src/core/utils.js

```diff
@@ -305,3 +293,53 @@ export function render404IfConfigKeyIsFalse(
     return <Component {...props} />;
   };
 }
+
+export function getCompabilityVersions({ addon, clientApp } = {}) {
+  let maxVersion = null;
+  let minVersion = null;
+  if (
+    addon && addon.current_version && addon.current_version.compatibility &&
+    addon.current_version.compatibility[clientApp]
+  ) {
+    maxVersion = addon.current_version.compatibility[clientApp].max;
+    minVersion = addon.current_version.compatibility[clientApp].min;
+  }
+
+  return { maxVersion, minVersion };
+}
+
+export function isCompatibleWithUserAgent({
+  userAgent, maxVersion, minVersion,
+} = {}) {
+  // If the userAgent is false-y, we'll assume it isn't valid.
+  if (!userAgent) {
+    return false;
+  }
+
+  const { browser, os } = UAParser(userAgent);
+
+  // We need a Firefox browser compatible with add-ons (Firefox for iOS does
+  // not currently support add-ons).
+  if (browser.name === 'Firefox' && os.name !== 'iOS') {
+    // Do version checks, if this add-on has minimum or maximum version
+    // requirements.
+    // The mozilla-version-comparator API is quite strange; a result of
+    // `1` means the first argument is higher in version than the second.
+    if (maxVersion && mozCompare(browser.version, maxVersion) === 1) {
+      return false;
+    }
+
+    // A result of `-1` means the second argument is a lower version than the
+    // first.
+    if (minVersion && mozCompare(browser.version, minVersion) === -1) {
+      return false;
+    }
+
+    // If we made it here we're compatible (yay!)
+    return true;
+  }
+
+  // This means the client is Firefox for iOS or not Firefox, so we mark
+  // the user agent as incompatible.
+  return false;
```

Comment:
Passing `i18n` to the function doesn't seem so bad. I think returning error codes seems ok too but it would be more work for you since you'd have more tests to write.